### DISCUSSION TITLE
how scittle handle comment line

### DISCUSSION THREAD

**Discussion by kangaroolab** [borkdude/scittle#85] [General]
it seems scittle doesn't handle comment line (see below). if I run following code with scittle, it doesn't work. any insight how to go about this?

```
;; Booleans

(def a-boolean true)

;; Strings

(def a-string "Hello!")

;; Regular Expressions

(def a-regexp #"\d{3}-?\d{3}-?\d{4}")

;; Numbers

(def a-number 1)
```

**Comment by borkdude** [comment]
Can't reproduce this. What is the error you're seeing?

**Comment by kangaroolab** [comment]
Two observations:

- if it starts with comment like above, it doesn’t run but stop at first line;
- If comment is in the middle of the code, it throw EOF error

**Comment by borkdude** [comment]
Please provide a full repro in a GitHub repro since I wasn’t able to
reproduce your problem

https://www.michielborkent.nl
https://www.eetvoorjeleven.nu

**Comment by borkdude** [comment]
Github repo I mean

https://www.michielborkent.nl
https://www.eetvoorjeleven.nu

**Comment by kangaroolab** [comment]
see [test](https://flems.io/#0=N4IgzgpgNhDGAuEAmIBcIB0ALeBbKIANCLAPYB2YpMaJ1UAhgA6QrFiwBO9aA2gAyEAjAE4MANgDMQgEwAWABxyA7HNlD+IuQF1iAMwCWMMH1DkGuCLWx4CxMuUSPaAHg6cDTeAAIwnWAC8ADogOPAsqAD0kbBI5BgAVmBI0AYAbpwY5BDwkeRMuJEcBvDwMAAC-BIYQsqRSAZgucWlMIlgId7wAJ5MEMEgiAAeuQkMaQzunvAhAHwuRVzTs0Hkqy4Nad4GSAPIJaScnU3dMANYEAYA5jio3pL8-ExDANzeAEaHKZx3Qs8vnQcTkQDXgDHeZxC8E4AFcIHN1psVuRvN4Xm8AMJQUgJGGcCAAZSWXm8ZEsYG8pBhPlIei6Fw+pCG3gA7iUsPSIN4YWAYQwoNzIHoYQKkAwwd4oCUIJx+WAMOt6ulkRtlatUQAKFJ0xBNbwhXBgK4hACUiqRiLVKK6ECa5qt9rSKve1PgFEp5FgUtgAGsBrDyBqzSBkaiA4qXaUKCqpl5Q95hZ74AZ3QGg95gOrUfqfN4JpxfHgfAFvEhSLAYZZHBgAI5wzjdAnQOBuzgagDkAGJ9q32yaMMN4BiKMCAdbUUFc01cMXCzOMPimIxYBANZENUEgkdVgAfTfkPdbk2RK64Qj6kCmsfZnOo6HdDNZm836LeYcpeni7y4a44encFlvAYFEICGFcvBTNZx2fVF828fFeSgWcWjKCAMDIfEMAgCYoAAfSaDxyCuDVp3gE1rxg7MHCoNpsWIhCRTIiiYIAX1JcVYA5DUIBNR9oJg18AAlgKQGBOW8UDwOTCgn0o6JqOoNC6O4jByTABgrh45jKIU2jSGItC1I0rTZJvFjZMnczyAWWN4FmIhwGbBBIJMdAhCEVB+BAFjCDMCwrHQdoHKBCBnHQbzfJAcxLGsWAwBMewR1C+BaG87QWKAA) 

if you click run, the output is null; if you remove the comment line on the top, it returns msg.

another test you can try is put the comment line in between code (e.g within a function), which will throw out EOF error

**Reply by borkdude** [reply]
You replace newlines with empty strings. Then you end up with something like this:

";; foo (+ 1 2 3)"

so all the code is on the same line and everything is a comment. Just don't replace those newlines and it will work like you expect.
